### PR TITLE
[TD]Balloon origin drag modifiers (fix #15388)

### LINKED ISSUE #15388: Techdraw: Balloon annotation pointer does not track reference in moves
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

In TechDraw, create a balloon annotation, then drag the created number. The tip of the arrow should keep pointing at the same spot in the drawing that was selected originally. Instead, the entire arrow moves with the balloon.

Rename [BrokenBalloon.zip](https://github.com/user-attachments/files/16195229/BrokenBalloon.zip) to BrokenBalloon.FCStd, go to the Page, grab the **1**, and drag it. The arrow point will not stay attached to the cube vertex; instead, the arrow keeps its shape and the point of the arrow will point at nothing.

### Full version info

```shell
OS: Debian GNU/Linux 12 (bookworm) (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38126 (Git)
Build type: Unknown
Branch: main
Hash: ac2b65c37b18617101822e5ece06f3c9b21375cd
Python 3.11.2, Qt 5.15.8, Coin 4.0.0, Vtk 9.1.0, OCC 7.6.3
Locale: English/United States (en_US)
Installed mods: 
  * Assembly4 0.50.4
  * Curves 0.6.13
  * sheetmetal 0.3.0
  * ExplodedAssembly
  * fasteners 0.5.24
  * MeshRemodel 1.8919.0
  * freecad.gears 1.0.0
  * Pyramids-and-Polyhedrons
  * dxf_library
  * lattice2 1.0.0
  * boltsfc 2022.11.5
  * in3dca-freegrid 2.0.0
  * parts_library
  * Defeaturing 1.2.0
  * A2plus 0.4.60n
  * CurvedShapes 1.0.4
  * Manipulator 1.5.0
  * Pyramids-and-Polyhedrons.backup1696298372.5270553 (Disabled)
  * Assembly3 0.12.2
  * DynamicData 2.62.0
```

### Subproject(s) affected?

Techdraw

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by WandererFan**
This PR implements a fix for issue #15388.

The change includes 
- changes to obscure variable names
- replacement of hard coded modifier key definitions with preferences
- changes to the handling of modifiers in QGIBalloonLabel
- support functions added to QGSPage
- changes to advanced preference tab.

AdvancedTab_current
AdvancedTab_proposed

### CODE REVIEW

**Comment by benj5378** [comment]
> changes to obscure variable names

😍

File: src/Mod/TechDraw/Gui/QGSPage.cpp [hunk 1]
```diff
@@ -147,10 +138,47 @@ void QGSPage::mouseReleaseEvent(QGraphicsSceneMouseEvent * event)
...
+//! returns true if clicking on the item should clear the selection
+bool QGSPage::itemClearsSelection(int itemTypeIn)
+{
+    // type 13 is the itemUnderMouse on a page outside of any views. It is not
+    // the template or background or foreground.  QGraphicsItem type = 13 is not
+    // documented and not found in QGraphicsItem.h.
+    const std::vector<int> ClearingTypes { 13,                        // MysteryType
+                                     QGraphicsItem::UserType + 150,   // QGITemplateType
+                                     QGraphicsItem::UserType + 151,   // QGIDrawingTemplateType
+                                     QGraphicsItem::UserType + 153 }; // QGISVGTemplateType
```
Comment (benj5378) [inline src/Mod/TechDraw/Gui/QGSPage.cpp:152]: To do; move type definitions to global file

Comment (WandererFan) [reply src/Mod/TechDraw/Gui/QGSPage.cpp:152]: Yeah, that's been on my list forever.

**Comment by chennes** [comment]
I see this is marked at 1.0 right now, but it appears to contain significant translatable string changes -- @WandererFan is it your plan to merge prior to 1.0, or is this really a post-1.0 PR?

**Comment by WandererFan** [comment]
The string changes are all for the new section on the advanced tab.  
It would be good to get this into 1.0 since if multiselection is enabled, balloon drag is broken.  There is a work around (turn off multiselection), but it is inconvenient.

**Review by chennes — COMMENTED** [review]
The way this is currently written, if I choose a combination that happens to be the default (Ctrl at the moment), the fact that I have explicitly set that is lost, and the "Use default" checkbox is selected. Is this the intended behavior? If so, why have the "Use default" option at all, instead of just having Ctrl be selected?

**Comment by WandererFan** [comment]
Without the "use default" option, you would have to know that ctrl is the default in order to revert your settings.  With an explicit "use default" you don't need to know the internals.

**Comment by chennes** [comment]
OK for me if it's OK for @FreeCAD/design-working-group .

**Comment by yorikvanhavre** [comment]
backported

**Pull request merged**